src/ui/components/search/searchcomponent.js

```diff
@@ -626,6 +631,7 @@ export default class SearchComponent extends Component {
    * @returns {Promise} A promise that will perform the query and update globalStorage accordingly.
    */
   search (query, searchOptions) {
+    this.onSubmit(query);
```

Comment:
I think this method is also called for searches on load, is that okay?

Reply:
The component also has an `onQuerySubmit` function. I believe that's fired only when the component's `form` is submitted, the search button is clicked, or a user hits return in the input. In short, I think it's only ever fired for searches caused by explicit user interaction. If you don't want `onSubmit` fired for searches on load,`onQuerySubmit` may be the place to put it.  

The `onQuerySubmit` function isn't exposed publicly. In the code, though, it may cause a bit of confusion to have that method and something called `onSubmit`. What do you think? One idea: we could call it `onSubmit` publicly, but name it something else in the component itself (i.e. `this.blah = config.onSubmit`)

Reply:
Yeah that totally makes sense to me! I combined this with the other feedback about nesting these commits under a "customHooks" top-level config option. In the code now, it's `this.customHooks.onSubmit()` which helps with the ambiguity. I also moved it to the `onQuerySubmit` function